sympy/geometry/line.py

```diff
@@ -1097,10 +1097,9 @@ def coordinate(self, other):
             if Point.is_collinear(self.p1, self.p2, other):
                 if self.p1.x != self.p2.x:
                     ti = (other.x - self.p1.x)/(self.p2.x - self.p1.x)
-                else:
+                elif self.p1.y != self.p2.y:
                     ti = (other.y - self.p1.y)/(self.p2.y - self.p1.y)
-                if ti.is_number:
-                    return ti
+                return ti
```

Comment:
If the two points coincide, `ti` will be undefined. It is probably better to replace each `ti =` by `return` and omit this line.